examples/calculator/lib/main.dart

```diff
@@ -0,0 +1,261 @@
+  // when the user hits the DEL key.
+  final List<CalcExpression> _expressionStack = <CalcExpression>[];
+  CalcExpression _expression = new CalcExpression.Empty();
+
+  // Make |expression| the current expression and push the previous current
+  // expression onto the stack.
+  void pushExpression(CalcExpression expression) {
+    _expressionStack.add(_expression);
+    _expression = expression;
+  }
+
+  // Pop the top expression off of the stack and make it the current expression.
+  void popCalcExpression() {
+    if (_expressionStack.length > 0) {
+      _expression = _expressionStack.removeLast();
+    } else {
+      _expression = new CalcExpression.Empty();
+    }
+  }
+
+  // Set |resultExpression| to the currrent expression and clear the stack.
+  void setResult(CalcExpression resultExpression) {
+    _expressionStack.clear();
+    _expression = resultExpression;
+  }
+
+  void onNumberTap(int n) {
+    CalcExpression expression = _expression.appendDigit(n);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onPointTap() {
+    CalcExpression expression = _expression.appendPoint();
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onPlusTap() {
+    CalcExpression expression = _expression.appendOperation(Operation.Addition);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onMinusTap() {
+    CalcExpression expression = _expression.appendMinus();
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onMultTap() {
+    CalcExpression expression =
+        _expression.appendOperation(Operation.Multiplication);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onDivTap() {
+    CalcExpression expression = _expression.appendOperation(Operation.Division);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onEqualsTap() {
+    CalcExpression resultExpression = _expression.computeResult();
+    if (resultExpression != null) {
+      setState(() {
+        setResult(resultExpression);
+      });
+    }
+  }
+
+  void onDelTap() {
+    setState(() {
+      popCalcExpression();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        appBar: new AppBar(title: new Text('Calculator')),
+        body: new Column(children: <Widget>[
+          // Give the key-pad 3/5 of the vertical space and the display
+          // 2/5.
+          new CalcDisplay(2, _expression.toString()),
+          new KeyPad(3, calcState: this)
+        ]));
+  }
+}
+
+class CalcDisplay extends StatelessWidget {
+  CalcDisplay(this._flex, this._contents);
```

Comment:
for widely-used widgets, we have the convention of always exposing a Key argument and having all the arguments be named. But you don't have to do that here since they are all internal widgets.